datasources/src/main/scala/com/normation/plugins/datasources/Repository.scala

```diff
@@ -168,30 +168,30 @@ class DataSourceRepoImpl(
   /*
    * Be careful, ALL modification to datasource must be synchronized
    */
-  private[this] object datasources extends AnyRef {
-    private[this] val semaphore   = Semaphore.make(1).runNow
-    private[this] val internalRef = Ref.make(Map[DataSourceId, DataSourceScheduler]()).runNow
+  private[datasources] object datasources extends AnyRef {
+    private[this] val semaphore          = Semaphore.make(1).runNow
+    private[datasources] val internalRef = Ref.make(Map[DataSourceId, DataSourceScheduler]()).runNow
 
     // utility methods on datasources
     // stop a datasource - must be called when the datasource still in "datasources"
-    private[this] def stop(id: DataSourceId) = {
+    private[datasources] def stop(id: DataSourceId) = {
       DataSourceLoggerPure.debug(s"Stopping data source with id '${id.value}'") *>
-      internalRef.get.map(_.get(id) match {
-        case None      => DataSourceLogger.trace(s"Data source with id ${id.value} was not found running")
+      internalRef.get.flatMap(_.get(id) match {
```

Comment:
this is the problem. `dss.cancel()` was never evaluated. It dates from rudder 6.0 !

Reply:
It made an IOResult [IOResult[Unit]] ?

Reply:
Maybe we could add return type of stop method